secretstores/akeyless/akeyless.go

```diff
@@ -0,0 +1,539 @@
+	}
+
+	a.logger.Debugf("getting secret value for '%s' (type %s)...", req.Name, secretType)
+
+	secretValue, err := a.GetSingleSecretValue(ctx, req.Name, secretType)
+	if err != nil {
+		return secretstores.GetSecretResponse{}, errors.New(err.Error())
+	}
+	a.logger.Debugf("secret '%s' value: %s", req.Name, secretValue[:3]+"[REDACTED]")
+
+	// Return the secret in the expected format
+	return GetDaprSingleSecretResponse(req.Name, secretValue)
+}
+
+// BulkGetSecret retrieves all secrets in the store and returns a map of decrypted string/string values.
+// The method performs the following steps:
+// 1. Recursively list all items in Akeyless
+// 2. Filter out inactive/failing secrets
+// 3. Separate items by type since only static secrets are supported for bulk get
+// 4. Get secret values concurrently, each item type in a separate goroutine
+func (a *akeylessSecretStore) BulkGetSecret(ctx context.Context, req secretstores.BulkGetSecretRequest) (secretstores.BulkGetSecretResponse, error) {
+	if a.v2 == nil {
+		return secretstores.BulkGetSecretResponse{}, errors.New("akeyless client not initialized")
+	}
+
+	// initialize response
+	response := secretstores.BulkGetSecretResponse{
+		Data: make(map[string]map[string]string),
+	}
+
+	// For bulk get, we need to list all secrets first
+	a.logger.Debug("listing items from / path...")
+	listItems, err := a.listItemsRecursively(ctx, "/")
+	if err != nil {
+		return response, fmt.Errorf("failed to list items from Akeyless: %w", err)
+	}
+
+	// if no items returned, return empty response
+	if len(listItems) == 0 {
+		a.logger.Debug("no items returned from / path")
+		return response, nil
+	}
+
+	// filter out inactive secrets
+	a.logger.Debugf("filtering out inactive secrets, %d items before filtering", len(listItems))
+	listItems = a.filterInactiveSecrets(listItems)
+	a.logger.Debugf("filtering out inactive secrets, %d items after filtering", len(listItems))
+
+	// separate items by type since only static secrets are supported for bulk get
+	staticItems, dynamicItems, rotatedItems := a.separateItemsByType(listItems)
+	a.logger.Infof("%d items returned (static: %d, dynamic: %d, rotated: %d)", len(listItems), len(staticItems), len(dynamicItems), len(rotatedItems))
+
+	// listItems can get quite large, so we don't need all item details, we can use the item names instead
+	// and free memory
+	listItems = nil
+	staticItemNames := GetItemNames(staticItems)
+	dynamicItemNames := GetItemNames(dynamicItems)
+	rotatedItemNames := GetItemNames(rotatedItems)
+	a.logger.Debugf("static items: %v", staticItemNames)
+	a.logger.Debugf("dynamic items: %v", dynamicItemNames)
+	a.logger.Debugf("rotated items: %v", rotatedItemNames)
+
+	haveStaticItems := len(staticItemNames) > 0
+	haveDynamicItems := len(dynamicItemNames) > 0
+	haveRotatedItems := len(rotatedItemNames) > 0
+
+	secretResultChannels := make(chan secretResultCollection, boolToInt(haveStaticItems)+boolToInt(haveDynamicItems)+boolToInt(haveRotatedItems))
```

Comment:
Each goroutine may send many results, esp dynamic/rotated secrets, but this buffer size equals only the number of goroutines, not the number of results. The goroutines will just block on channel send when the buffer fills, but it's inefficient if large sets are fetched. Did you try making the buffer large enough for expected items or leave it unbuffered (so the consumer loop will drain continuously anyways)?
for ex:

```
secretResultChannels := make(chan secretResultCollection)
// or
secretResultChannels := make(chan secretResultCollection, len(listItems))
```